docs/Roadmap.md

```diff
@@ -36,35 +25,75 @@ instance. Care must be taken to keep the standalone version usable.
+### 2023 goals
+
+- **Modernising the codebase**. 2023 has to be the year of all Typescript. We need to continue pushing towards this goal
+  as we are currently blocked using old versions of some libraries due to JS/TS/node.js/ES6 modules conflicts. We're
+  still using an outdated and unsupported module loader and in order to move off it we need to update to a newer
+  loader...that itself doesn't seemingly doesn't support a mixture of typescript and javascript like we currently have.
+  In any case, moving to Typescript has improved our code quality and continues to find bugs, so we should absolutely
+  port to it for this reason alone.
+- **Improved compiler installation directory handling**. This project started in 2022 but left to rot a little, but our
+  existing approach to sharing the 1700+ compilers and libraries between running instances is not scaling well. Behind
+  the scenes we use NFS, and then for performance mount some compilers as read-only squashfs images from NFS. That
+  worked well until we had 1000+ squashfs images to mount, one per compiler. A project to administrate and update a more
+  layered squashfs image containing large numbers of unchanged compilers (think `docker` image layers) was started in
+  2022, but fell behind due to so many other issues, and Matt being pretty rubbish at prioritising things. As nodes are
+  now taking an awful long time to start up, this is becoming a problem.
+- **Improved deployment flow**. The work to deploy Compiler Explorer has increased. At one point I couldn't get a site
+  release out in the length of my train commute to work, which was a tipping point. I took a week off work in 2022 to
+  fix that (amongst other things), but it's still a long and more-tedious-than-I-would-like process. With the recent
+  addition of a GPU node, it's also more tricky than before to keep things updated. I'd like to find a way to get
+  instances to start up without the need to pre-cache the compilation versions (a step we currently require to make
+  startup fast enough). Ideally an update should be able to be directly applied to our staging setup from GitHub, and
+  then promoted to production once things look good.
+- **More resource pools**. Now we have two pools of servers - those with GPUs and those without - it would be great to
+  support other server types too. For example, we could run our own Windows instances, or ARM-based instances too. This
+  would allow us to natively run ARM code, or run `clang-cl` on a Windows instance we administrate and can make changes
+  to. We probably need a better way of expressing the various pool types, and certainly this would need better
+  deployment support (see above).
 
 ### Prior years' goals
 
+#### 2022
+
+Sadly I set no goals in 2022. I decided not to continue pursuing the login server stuff. It would be lovely but the
+risks of data leakage/privacy violations, plus the work needed to get it running, seem to outweigh the benefits.
```

Comment:
Can't blame you for not pursuing the login idea for now, there's definitely a lot of things to consider!

Reply:
I think this is still a thing that I want to pursue, as it's something that adds value to the website

Reply:
In the future, for sure. I'm trying to be pragmatic about the direction of things _I_ nave control over: though I'm happy to include goals like this too. My concern is that ultimately the burden of legal and reputational stuff falls on me, and so we'll have to be careful about that side of things.

Reply:
And to be 100% nothing here precludes looking in to things!